safety_prototype/opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -0,0 +1,42 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This an implementation of the API where every function or method is safe.
+
+Any call to a function or method defined here passes its arguments to a
+corresponding underlying SDK function or method. In this way, objects defined
+in this module act as proxies and (with the SDK setting function) are the only
+objects the user has contact with.
+"""
+
+from opentelemetry._safety import _safe_function
+from opentelemetry.configuration import _get_sdk_module
+from opentelemetry.trace.api import Class0
+
+
+@_safe_function(0.0)
+def function(a: int, b: int) -> float:
+    return _get_sdk_module("trace").function(a, b)
```

Comment:
> pylint/mypy/IDEs can check things like that your import paths are correct or that you implement an ABC correctly. In this case, it's impossible to tell these tools "this package must mirror all of the interfaces of OpenTelemetry API".

Ok, but I never expected or intended for the static checkers to be the ones responsible of checking that the SDKs are compliant with the API.

This is a very OpenTelemetry-specific requirement, I don't think static checkers or the `typing` module to be designed with this in mind, so this approach looks like forcing our design to work with a tool that was never intended to check this kind of things.

The approach here is for the API to have functions and abstract classes that the SDK has to implement. We can provide a list of the functions and classes so that the SDK implementations know what they have to implement and they can check that their SDKs implement them fully.